library/src/main/java/com/github/pwittchen/reactivesensors/library/ReactiveSensors.java

```diff
@@ -134,7 +155,72 @@ public boolean hasSensor(int sensorType) {
         }
       }
     }, strategy).doOnCancel(new Action() {
-      @Override public void run() throws Exception {
+      @Override
+      public void run() throws Exception {
+        sensorManager.unregisterListener(listener);
+      }
+    });
+  }
+
+  /**
+   * Returns RxJava Observable, which allows to monitor hardware sensors as a stream of
+   * ReactiveSensorEvent object with defined sampling period
+   *
+   * @param samplingPeriodInUs sampling period in microseconds,
+   * @param handler the Handler the sensor events will be delivered to, use default if it is null
+   * you can use predefined values from SensorManager class with prefix SENSOR_DELAY
+   * @param strategy BackpressureStrategy for RxJava 2 Flowable type
+   * @param sensorTypes sensor types array from Sensor class from Android SDK
+   * @return RxJava Observable with ReactiveSensorEvent
+   */
+  public Flowable<ReactiveSensorEvent> observeSensors(final int samplingPeriodInUs,
+      final Handler handler,
+      final BackpressureStrategy strategy, int... sensorTypes) {
+    StringBuilder errorMessage = null;
+    for (int sensorType : sensorTypes) {
+      if (!hasSensor(sensorType)) {
+        if (errorMessage == null) {
```

Comment:
I think, it can be simplified like that:

```java
if(errorMessage == null) {
  errorMessage = new StringBuilder("Following sensors are not available on current device: ");
}

errorMessage.append(sensorType);
errorMessage.append(" ");
```

In such case, `else` block could be skipped.